pyEpiabm/pyEpiabm/sweep/update_place_sweep.py

```diff
@@ -20,25 +20,90 @@ def __call__(self, time: int):
         :type time: int
         """
         # Double loop over the whole population, clearing places
-        # and refilling them.
+        # of the variable population and refilling them.
+
+        # Can call a this line if being called in from file etc.
+        # place_params = Parameters.instance().place_params
         for cell in self._population.cells:
             for place in cell.places:
-                place.empty_place()
-                # Possibly want to later differentiate between
-                # the fixed population of a place ie staff, and
-                # the variable population ie customers.
-
-                # Ensure that the number of people put in the place
-                # is at most its capacity or the total number of
-                # people in the cell.
-                max_capacity = np.minimum(place.max_capacity,
-                                          len(place.cell.persons))
-
-                new_capacity = random.randint(1, max_capacity)
-                count = 0
-                while count < new_capacity:
-                    i = random.randint(1, len(place.cell.persons))
-                    # Checks person is not already in the place.
-                    if place.cell.persons[i-1] not in place.persons:
-                        place.add_person(place.cell.persons[i-1])
-                        count += 1
+                if place.place_type.value == 1:  # HOTEL
+                    place.empty_place()
+                    self.update_place_group(place)
+                    # not sure how to handle this yet so
+                    # just going to completely update each time
+                    # as in the previous code
+                elif place.place_type.value == 1:  # CAREHOME
+                    if not place.initialised:
+                        # Initialise the fixed population of workers
+                        max_cap = 10  # for example, would actually come in
+                        # from a dictionary
+                        self.update_place_group(place, group_index=0)
+                        max_residents = 100
+                        self.update_place_group(place,
+                                                max_capacity=max_residents,
+                                                group_index=1)
+                        place.initialised = True
+
+                elif place.place_type.value == 3:  # RESTAURANT
+                    if not place.initialised:
+                        # Initialise the fixed population
+                        max_cap = 10
+                        self.update_place_group(place, max_capacity=max_cap,
+                                                group_index=0)
+                        place.initialised = True
+                    # Variable population is people not in the fixed pop.
+                    # Changed at each timestep
+                    place.empty_place(1)
```

Comment:
```suggestion
                    place.empty_place(person_groups=[1])
```
Readability suggestion, as its really not clear what the argument is otherwise from the function name. Also think you need to pass a list not a number?